src/structs.jl

```diff
@@ -229,6 +236,29 @@ read(::ArrayType, buf, pos, len, b, ::Type{Tuple}, ::Type{eT}; kw...) where {eT}
     invalid(error, buf, pos, T)
 end
 
+@inline readjsonlines(buf, pos, len, b, ::Type{T}; kw...) where {T} = readjsonlines(buf, pos, len, b, T, Base.IteratorEltype(T) == Base.HasEltype() ? eltype(T) : Any; kw...)
+@inline readjsonlines(buf, pos, len, b, ::Type{T}, ::Type{eT}; kw...) where {T, eT} = readjsonlinesarray(buf, pos, len, b, T, eT; kw...)
+
+@inline function readjsonlinesarray(buf, pos, len, b, ::Type{T}, ::Type{eT}; kw...) where {T, eT}
+    vals = Vector{eT}(undef, 0)
+    if pos > len
+        @goto done
+    end
+    while true
+        # positioned at start of value
```

Comment:
We should add in a `@wh` here, just to ensure any spaces/tabs are consumed before starting to read the struct (reading a struct assumes we're positioned at a `{` character).

Reply:
```suggestion
        # positioned at start of value
        @wh
```